skrf/networkSet.py

```diff
@@ -209,6 +210,8 @@ def __init__(self, ntwk_set: list | dict = None, name: str = None):
             self.__add_a_element_wise_method('plot_'+network_property_name)
             self.__add_a_element_wise_method('plot_s_db')
             self.__add_a_element_wise_method('plot_s_db_time')
+            self.__add_a_plot_violin(network_property_name)
```

Comment:
To be honest I don't like the autogenerated method stuff, although I'm aware that we used this way several times in this repository. It's hard to remove existing methods as we would break most existing code, but this isn't the case here.

These methods were not recogniced by any type checker and won't have autocompletion therefore.

I'd prefer to have _one_ method called `plot_method(self, ...)` with a similar attribute like this one?
https://github.com/scikit-rf/scikit-rf/blob/8c5729dd304a3d668c2b4058330cd8e85ac99fe1/skrf/network.py#L4661

@jhillairet what do you think?

Reply:
I agree, the lack of autocomplete is inconvenient and it is difficult to follow for development. It's also not clear what all the possible methods are in the api documentation. For now, I just decided to use the existing convention for consistency but I'm definitely open to other ideas.

With the single method approach, wouldn't the user have to pass in the property and function argument and need to know what the possible options are? Maybe something like enums would be a good way to implement that?

Reply:
> I'd prefer to have _one_ method called `plot_method(self, ...)` with a similar signature like the one in plotting.py
> 
> @jhillairet what do you think? If you are fine, I will not block :)

Sure, the lack of autocompletion is indeed inconvenient.What would you suggest for @ericwrice as a replacement?

Reply:
I would recommend leaving it as is for now to match the existing convention and then add in new methods for all plotting features that are type checker friendly. Eventually, the auto generated methods could then be deprecated if desired. 

Maybe something along the lines of this pseudo-code:

```python
from enum import StrEnum, auto


class PROPERTIES(StrEnum):
    s = auto()
    z = auto()
    y = auto()
    a = auto()
    h = auto()
    t = auto()

class FUNCTIONS(StrEnum):
    re = auto()
    im= auto()
    mag = auto()
    db = auto()
    db10 = auto()
    rad = auto()
    deg = auto()
    arcl = auto()
    rad_unwrap = auto()
    deg_unwrap = auto()
    arcl_unwrap = auto()
    vswr = auto()
    time = auto()
    time_db = auto()
    time_mag = auto()
    time_impulse = auto()
    time_step = auto()

# There would be similar implementations for other plots such as plot_smith(), plot_violin(), etc.
def plot_rect(self: Network, attribute: PROPERTIES, conversion: FUNCTIONS, *args, **kwargs):
    self.plot_attribute(attribute, conversion, *args, **kwargs)

# These would be used instead of plot_s_db()
plot_rect(PROPERTIES.s, FUNCTIONS.db)  # Autocomplete and type checkers should work
plot_rect('s', 'db')  # Strings could be used instead, though some type checkers will complain
```

Reply:
Even better than using `Enum` is the `Literal` type in this case. 
For the user it's used exactly like a string (no import needed) but type checkers recognice if a allowed string is used.
https://peps.python.org/pep-0586/#shortening-unions-of-literals

However this is only available from 3.8 up, but I think it's to move on in the next time anyway.

Reply:
This done for example in numpy's `fft` code, where you can choose `forward`, `backward` or `ortho` norm.

https://numpy.org/doc/stable/reference/generated/numpy.fft.fft.html#numpy.fft.fft
https://github.com/numpy/numpy/blob/v2.0.0/numpy/fft/_pocketfft.pyi#L15

Reply:
Yep, I like this suggestion. The 3.7 support was my concern with proposing this. I'll keep this on the list for future versions.